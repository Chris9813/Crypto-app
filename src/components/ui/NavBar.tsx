
import { memo } from 'react';
import { Link, NavLink } from 'react-router-dom'

export const Navbar = memo(() => {
    console.log("holi");
    
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/coins"
            >
                Crypto
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/coins"
                    >
                        Coins
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/exchanges"
                    >
                        Exchanges
                    </NavLink>
                    
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ml-auto">
                </ul>
            </div>
        </nav>
    )
})