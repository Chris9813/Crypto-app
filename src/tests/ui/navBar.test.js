import { mount, shallow } from "enzyme";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { Navbar } from "../../components/ui/NavBar";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  NavLink: () => mockNavigate,
}));

describe("Pruebas es en componente navBar", () => {
  const wrapper = shallow(<Navbar />);

  test("debe de mostrar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
