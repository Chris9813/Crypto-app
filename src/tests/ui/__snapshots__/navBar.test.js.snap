// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es en componente navBar debe de mostrar correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <Link
    className="navbar-brand"
    to="/coins"
  >
    Crypto
  </Link>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/coins"
      >
        Coins
      </NavLink>
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/exchanges"
      >
        Exchanges
      </NavLink>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    />
  </div>
</nav>
`;
